env0/resource_environment_scheduling.go

```diff
@@ -0,0 +1,128 @@
+package env0
+
+import (
+	"context"
+	"github.com/adhocore/gronx"
+	. "github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironmentScheduling() *schema.Resource {
+
+	validateCronExpression := func(i interface{}, path cty.Path) diag.Diagnostics {
+		expr := i.(string)
+		parser := gronx.New()
+		isValid := parser.IsValid(expr)
+
+		if isValid != true {
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity:      diag.Error,
+					Summary:       "Invalid cron expression",
+					AttributePath: path,
+				}}
+		}
+
+		return nil
+	}
+
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentSchedulingCreateOrUpdate,
+		ReadContext:   resourceEnvironmentSchedulingRead,
+		UpdateContext: resourceEnvironmentSchedulingCreateOrUpdate,
+		DeleteContext: resourceEnvironmentSchedulingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"environment_id": {
+				Type:        schema.TypeString,
+				Description: "the environment's id",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"destroy_cron": {
+				Type:             schema.TypeString,
+				Description:      "Cron expression for scheduled destroy of the environment",
+				AtLeastOneOf:     []string{"destroy_cron", "deploy_cron"},
+				Optional:         true,
+				ValidateDiagFunc: validateCronExpression,
+			},
+			"deploy_cron": {
+				Type:             schema.TypeString,
+				Description:      "Cron expression for scheduled deploy of the environment",
+				AtLeastOneOf:     []string{"destroy_cron", "deploy_cron"},
+				Optional:         true,
+				ValidateDiagFunc: validateCronExpression,
+			},
+		},
+	}
+}
+
+func resourceEnvironmentSchedulingRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(ApiClientInterface)
+
+	environmentId := d.Id()
+
+	environmentScheduling, err := apiClient.EnvironmentScheduling(environmentId)
+
+	if err != nil {
+		return diag.Errorf("could not get environment scheduling: %v", err)
+	}
+
+	if environmentScheduling.Deploy != nil {
+		d.Set("deploy_cron", environmentScheduling.Deploy.Cron)
+	} else {
+		d.Set("deploy_cron", "")
```

Comment:
why you need to set that?
isn't that empty string by default? or why does it matter?

Reply:
That's how you handle drift detection. In case something removed one of the crons, the read operation needs to be update the state accordingly